app/ui-react/packages/ui/src/Customization/apiClientConnectors/create/ApiConnectorCreatorSecurityForm.tsx

```diff
@@ -1,92 +1,84 @@
 import * as React from 'react';
+import validateSecurity from './ApiConnectorValidationRules';
 
-export interface IApiConnectorCreatorSecurityFormChildrenProps {
-  authUrl?: string;
-  handleChangeAuthUrl: (params: string) => void;
-  handleChangeSelectedType: (params: string) => void;
-  handleChangeTokenUrl: (params: string) => void;
-
-  /**
-   * Access token, required for OAuth 2.0.
-   */
-  initialAccessTokenUrl?: string;
-  /**
-   * Used specifically for determining the default type, mostly used
-   * for None and Basic types.
-   */
-  initialAuthenticationType?: string;
+/**
+ * Customizable properties in API Client Connector wizard
+ */
+export interface ICreateConnectorPropsUi {
+  addTimestamp?: boolean;
+  addUsernameTokenCreated?: boolean;
+  addUsernameTokenNonce?: boolean;
+  authenticationType?: string;
+  authorizationEndpoint?: string;
+  basePath?: string;
+  connectorTemplateId?: string;
+  description?: string;
+  host?: string;
+  icon?: string;
+  name?: string;
+  password?: string;
+  passwordType?: string;
   /**
-   * Authorization URL, required for OAuth 2.0.
+   * portName & serviceName
+   * are used for SOAP documents
    */
-  initialAuthorizationUrl?: string;
-  selectedType?: string;
-  tokenUrl?: string;
-  valid: boolean;
+  portName?: string;
+  serviceName?: string;
+  specification?: string;
+  tokenEndpoint?: string;
+  username?: string;
+}
+
+export interface IApiConnectorCreatorSecurityFormChildrenProps {
+  errors?: any;
+  handleChange?: (param?: any, event?: any) => void;
+  handleSubmit?: (param?: any) => void;
+  values: ICreateConnectorPropsUi;
 }
 
 export interface IApiConnectorCreatorSecurityFormProps {
-  /**
-   * Access token, required for OAuth 2.0.
-   */
-  initialAccessTokenUrl?: string;
-  /**
-   * Used specifically for determining the default type, mostly used
-   * for None and Basic types.
-   */
-  initialAuthenticationType?: string;
-  /**
-   * Authorization URL, required for OAuth 2.0.
-   */
-  initialAuthorizationUrl?: string;
+  defaultValues?: any;
```

Comment:
No chance we could use a concrete interface or [union](https://www.tutorialspoint.com/typescript/typescript_union.htm) here?